cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -697,7 +697,7 @@ impl ABISig {
 }
 
 /// ABI object for a function body.
-pub struct ABICalleeImpl<M: ABIMachineSpec> {
+pub struct AbiCallee<M: ABIMachineSpec> {
```

Comment:
Can we keep the capitalization `ABICallee`? I know there are varying opinions on how camelcase interacts with acronyms; but I find `Abi` for the acronym `ABI` a little unnatural to read. (Sorry for the bikeshed color suggestion!)

Reply:
I changed it because it is Rust's naming style to only capitalize the first letter of an acronym in a type (consistent with how you wouldn't add underscores between each letter in the acronym in snake case):

> In <code class="hljs">UpperCamelCase</code>, acronyms and contractions of compound words count as one word: use <code class="hljs">Uuid</code> rather than <code class="hljs">UUID</code>, <code class="hljs">Usize</code> rather than <code class="hljs">USize</code> or <code class="hljs">Stdin</code> rather than <code class="hljs">StdIn</code>. In <code class="hljs">snake_case</code>, acronyms and contractions are lower-cased: <code class="hljs">is_xid_start</code>.

https://rust-lang.github.io/api-guidelines/naming.html

And also because I happen to find it more readable that way as well (no mixing up the "C" in "Callee" as part of the acronym). I plan on doing the rest in follow up PRs that I'm working on.

Reply:
When there's a collision between widespread naming conventions and readability, I like to have... both? :sweat_smile:

Is there an alternative name that's meaningful but avoids the "ABI" acronym entirely? Or perhaps, would `CalleeAbi` work as a compromise? Somehow that reads less badly to me.

Reply:
I guess I diverge from the above guidelines then in this particular area :-) This paragraph from the "About" page of that book seems relevant too:

> These are only guidelines, some more firm than others. In some cases they are vague and still in development. Rust crate authors should consider them as a set of important considerations in the development of idiomatic and interoperable Rust libraries, to use as they see fit. These guidelines should not in any way be considered a mandate that crate authors must follow, though they may find that crates that conform well to these guidelines integrate better with the existing crate ecosystem than those that do not.

I find `AbiCallee` looks wrong to me; an ABI is an ABI, not an Abi (or "abi"); I'd really prefer not to take the change. If it's confusing to me and not confusing to you, then there is likely a split among other readers as well. Given that it is currently this way and there is disagreement, could we keep with the current convention for this PR and then discuss separately? (I'd the same concern if the other names were also modified in this way.)

Reply:
I saw @jameysharp 's reply after writing mine -- but this brings to mind another option: just `Caller` and `Callee`? (or `CallerHandler` and `CalleeHandler`?)

Reply:
Ooh, I like `abi::Callee`.